test/sqllogictest/role_create.slt

```diff
@@ -0,0 +1,148 @@
+
+# This should fail, see https://github.com/MaterializeInc/materialize/issues/18960
+simple conn=regress_role_limited_admin,user=regress_role_limited_admin
+CREATE ROLE regress_nosuch_createdb CREATEDB;
+----
+COMPLETE 0
+
+# This should fail, see https://github.com/MaterializeInc/materialize/issues/18960
+simple conn=regress_role_limited_admin,user=regress_role_limited_admin
+CREATE ROLE regress_nosuch_createcluster CREATECLUSTER;
+----
+COMPLETE 0
+
+simple conn=regress_role_limited_admin,user=regress_role_limited_admin
+CREATE ROLE regress_role_limited;
+----
+COMPLETE 0
+
+# This should fail, see https://github.com/MaterializeInc/materialize/issues/18960
+simple conn=regress_role_limited_admin,user=regress_role_limited_admin
+ALTER ROLE regress_role_limited CREATEDB;
+----
+COMPLETE 0
+
+# This should fail, see https://github.com/MaterializeInc/materialize/issues/18960
+simple conn=regress_role_limited_admin,user=regress_role_limited_admin
+ALTER ROLE regress_role_limited CREATECLUSTER;
+----
+COMPLETE 0
+
+simple conn=regress_role_admin,user=regress_role_admin
+CREATE ROLE regress_createdb CREATEDB;
+----
+COMPLETE 0
+
+simple conn=regress_role_admin,user=regress_role_admin
+ALTER ROLE regress_createdb NOCREATEDB;
+----
+COMPLETE 0
+
+simple conn=regress_role_admin,user=regress_role_admin
+ALTER ROLE regress_createdb CREATEDB;
+----
+COMPLETE 0
+
+simple conn=regress_role_admin,user=regress_role_admin
+CREATE ROLE regress_createcluster CREATECLUSTER;
+----
+COMPLETE 0
+
+simple conn=regress_role_admin,user=regress_role_admin
+ALTER ROLE regress_createcluster NOCREATECLUSTER;
+----
+COMPLETE 0
+
+simple conn=regress_role_admin,user=regress_role_admin
+ALTER ROLE regress_createcluster CREATECLUSTER;
+----
+COMPLETE 0
+
+simple conn=regress_role_admin,user=regress_role_admin
+ALTER ROLE regress_createdb SUPERUSER;
+----
+db error: ERROR: SUPERUSER attribute is not supported, for more information consult the documentation at https://materialize.com/docs/sql/create-role/#details
+
+simple conn=regress_role_admin,user=regress_role_admin
+ALTER ROLE regress_createdb NOSUPERUSER;
+----
+db error: ERROR: SUPERUSER attribute is not supported, for more information consult the documentation at https://materialize.com/docs/sql/create-role/#details
+
+simple conn=regress_role_admin,user=regress_role_admin
+CREATE ROLE regress_createrole CREATEROLE;
+----
+COMPLETE 0
+
+simple conn=regress_role_admin,user=regress_role_admin
+CREATE ROLE regress_connection_limit CONNECTION LIMIT 5;
+----
+db error: ERROR: Expected end of statement, found CONNECTION
+
+simple conn=regress_role_admin,user=regress_role_admin
+CREATE ROLE regress_password_null PASSWORD NULL;
+----
+db error: ERROR: Expected end of statement, found PASSWORD
+
+simple conn=regress_role_admin,user=regress_role_admin
+CREATE ROLE regress_noiseword SYSID 12345;
+----
+db error: ERROR: Expected end of statement, found identifier "sysid"
```

Comment:
Can you add all the non-supported role attributes from https://www.postgresql.org/docs/current/sql-createrole.html to this test in order to confirm that they error out. This is to protect any from being accepted in the parser now or in the future without being tested properly -- and we have had cases in the past where some postgres syntax was accepted by the parser and then silently ignored by the rest of the code.